ecs-cli/modules/cli/compose/entity/service/service_test.go

```diff
@@ -601,3 +601,311 @@ func TestServiceRun(t *testing.T) {
+	upServiceWithNewTaskDefTest(t, cliContext, &config.CLIParams{}, &utils.ECSParams{}, expectedInput, existingService)
+}
+
+func TestUpdateExistingServiceWithNewHCGP(t *testing.T) {
+	// define test flag set
+	healthCheckGracePeriod := 200
+
+	flagSet := flag.NewFlagSet("ecs-cli-up", 0)
+	flagSet.String(flags.HealthCheckGracePeriodFlag, strconv.Itoa(healthCheckGracePeriod), "")
+	cliContext := cli.NewContext(nil, flagSet, nil)
+
+	// define existing service
+	serviceName := "test-service"
+	existingService := &ecs.Service{
+		TaskDefinition: aws.String("arn/test-task-def"),
+		Status:         aws.String("ACTIVE"),
+		DesiredCount:   aws.Int64(0),
+		ServiceName:    aws.String(serviceName),
+		LoadBalancers:  []*ecs.LoadBalancer{}, // LB required for HCGP, but not verified before calling client
+	}
+
+	// define expected client input given the above info
+	expectedInput := getDefaultUpdateInput()
+	expectedInput.serviceName = serviceName
+	expectedInput.healthCheckGracePeriod = aws.Int64(int64(healthCheckGracePeriod))
+
+	// call tests
+	upServiceWithCurrentTaskDefTest(t, cliContext, &config.CLIParams{}, &utils.ECSParams{}, expectedInput, existingService)
+	upServiceWithNewTaskDefTest(t, cliContext, &config.CLIParams{}, &utils.ECSParams{}, expectedInput, existingService)
+}
+
+func TestUpdateExistingServiceWithDesiredCountOverOne(t *testing.T) {
+	// define test values
+	existingDesiredCount := 2
+
+	flagSet := flag.NewFlagSet("ecs-cli-up", 0)
+	cliContext := cli.NewContext(nil, flagSet, nil)
+
+	// define existing service
+	serviceName := "test-service"
+	existingService := &ecs.Service{
+		TaskDefinition: aws.String("arn/test-task-def"),
+		Status:         aws.String("ACTIVE"),
+		DesiredCount:   aws.Int64(int64(existingDesiredCount)), // existing count > 1
+		ServiceName:    aws.String(serviceName),
+	}
+
+	// define expected client input given the above info
+	expectedInput := getDefaultUpdateInput()
+	expectedInput.serviceName = serviceName
+	expectedInput.count = *aws.Int64(int64(existingDesiredCount))
+
+	// call tests
+	upServiceWithCurrentTaskDefTest(t, cliContext, &config.CLIParams{}, &utils.ECSParams{}, expectedInput, existingService)
+	upServiceWithNewTaskDefTest(t, cliContext, &config.CLIParams{}, &utils.ECSParams{}, expectedInput, existingService)
+}
+
+func getDefaultUpdateInput() UpdateServiceParams {
+	return UpdateServiceParams{
+		deploymentConfig: &ecs.DeploymentConfiguration{}, //TODO: submit nil instead of empty obj on Update
+		count:            1,
+	}
+}
+
+func upServiceWithCurrentTaskDefTest(t *testing.T,
+	cliContext *cli.Context,
+	cliParams *config.CLIParams,
+	ecsParams *utils.ECSParams,
+	expectedInput UpdateServiceParams,
+	existingService *ecs.Service) {
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// setup expected task def
+	taskDefArn := "arn/taskDefinitionId"
+	if existingService != nil { // set to existing if provided
```

Comment:
tiniest of tiniest of nits: can the comment either be on a new line of at least inline with the following line? Something about comments coming after the opening brace makes my eye twitch...